.github/workflows/ci-testing-deploy.yml

```diff
@@ -1178,21 +1149,45 @@ jobs:
       - uses: codecov/codecov-action@v3.1.1
         with:
           flags: unittests #optional
-      - name: prepare codeclimate coverage file
-        run: |
-          curl --location https://codeclimate.com/downloads/test-reporter/test-reporter-0.10.3-linux-amd64 > ./cc-test-reporter
-          chmod +x ./cc-test-reporter && ./cc-test-reporter --version
-          ./cc-test-reporter format-coverage --input-type coverage.py --prefix=/ --output codeclimate.${{ github.job }}_coverage.json packages/simcore-sdk/coverage.xml
-      - name: store codeclimate coverage
-        uses: actions/upload-artifact@v3
-        with:
-          name: codeclimate-${{ github.job }}-coverage
-          path: codeclimate.${{ github.job }}_coverage.json
+
+  unit-tests:
+    # NOTE: this is a github required status check!
+    if: always()
+    needs:
+      [
+        unit-test-api,
+        unit-test-api-server,
+        unit-test-autoscaling,
+        unit-test-catalog,
+        unit-test-dask-sidecar,
+        unit-test-dask-task-models-library,
+        unit-test-datcore-adapter,
+        unit-test-director,
+        unit-test-director-v2,
+        unit-test-dynamic-sidecar,
+        unit-test-frontend,
+        unit-test-models-library,
+        unit-test-postgres-database,
+        unit-test-python-linting,
+        unit-test-service-integration,
+        unit-test-service-library,
+        unit-test-settings-library,
+        unit-test-simcore-sdk,
+        unit-test-storage,
+        unit-test-webserver-01,
+        unit-test-webserver-02,
+        unit-test-webserver-03,
+      ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: all unit tests were run or skipped
+        run: echo "all unit tests were run or skipped"
 
   integration-test-webserver-01:
+    needs: [changes, build-test-images]
+    if: ${{ needs.changes.outputs.anything-py == 'true' || github.event_name == 'push' }}
```

Comment:
For integration testing, it is always triggered if some python code changes? Is my interpretation correct?

Reply:
that is correct. it could be more granular, but I thought this would need too much maintenance as it would have to be in sync with the package dependencies.